src/construction.cpp

```diff
@@ -1377,13 +1377,13 @@ void check_constructions()
             debugmsg("Unknown skill %s in %s", c->skill.c_str(), display_name.c_str());
         }
         c->requirements.check_consistency(display_name);
-        if (!c->pre_terrain.empty() && !c->pre_is_furniture && termap.count(c->pre_terrain) == 0) {
+        if (!c->pre_terrain.empty() && !c->pre_is_furniture && !ter_str_id( c->pre_terrain ).is_valid() ) {
```

Comment:
> After that, all ids should be assumed to be valid.

Good point, but how to treat invalid identifiers apart from reporting? Leave them as they are, or replace with NULL_ID?

>The logic here is bad (but that is a obviously a pre-existing problem).

Totally agreed, these should be two independent structures, but this PR isn't addressing this problem (at least yet).